src/main/java/com/facebook/openwifirrm/modules/RRMScheduler.java

```diff
@@ -156,11 +169,101 @@ public void shutdown() {
 		}
 	}
 
+	/**
+	 * TODO [ZoneBasedRrmScheduling] remove this and used venue based config.
+	 * Synchronize triggers to the current topology, adding/updating/deleting
+	 * them as necessary. This updates {@link #scheduledDevices}.
+	 */
+	public void syncTriggersForDevices() {
+		Set<String> scheduled = ConcurrentHashMap.newKeySet();
+		Set<String> prevScheduled = new HashSet<>();
+		if (scheduledDevices != null) {
+			prevScheduled.addAll(scheduledDevices);
+		}
+
+		Map<String, Set<String>> topology = deviceDataManager.getTopologyCopy();
+
+		// Add new triggers
+		for (Map.Entry<String, Set<String>> entry : topology.entrySet()) {
+			for (String serialNumber : entry.getValue()) {
+				DeviceConfig config =
+					deviceDataManager.getDeviceConfig(serialNumber);
+				if (
+					config.schedule == null ||
+						config.schedule.cron == null ||
+						config.schedule.cron.isEmpty()
+				) {
+					continue; // RRM not scheduled
+				}
+
+				try {
+					CronExpression.validateExpression(config.schedule.cron);
+				} catch (ParseException e) {
+					logger.error(
+						String.format(
+							"Invalid cron expression (%s) for device %s",
+							config.schedule.cron,
+							serialNumber
+						),
+						e
+					);
+					continue;
+				}
+
+				// Create trigger
+				Trigger trigger = TriggerBuilder.newTrigger()
+					.withIdentity(serialNumber)
+					.forJob(job)
+					.withSchedule(
+						CronScheduleBuilder.cronSchedule(config.schedule.cron)
+					)
+					.build();
+				try {
+					if (!prevScheduled.contains(serialNumber)) {
+						scheduler.scheduleJob(trigger);
+					} else {
+						scheduler.rescheduleJob(trigger.getKey(), trigger);
+					}
+				} catch (SchedulerException e) {
+					logger.error(
+						"Failed to schedule RRM trigger for device: " +
+							serialNumber,
+						e
+					);
+					continue;
+				}
+				scheduled.add(serialNumber);
```

Comment:
Should we put this line in the try block? Only APs that are successfully scheduled are added.

Reply:
the loop will continue without getting to this point due to the `continue` on line 233